# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1328737453 18000
#      Wed Feb 08 16:44:13 2012 -0500
# Branch stable
# Node ID 06f706e92771f1491c1a3045bc670a57a6480a9d
# Parent  acc26b860afa76f3dc131975b0f29b0681fcfcd5
avoid profiler test failure on more systems

* profiler.cc (profile_data_accumulator::query_time): Declare dnow
volatile on all systems.

diff --git a/src/profiler.cc b/src/profiler.cc
--- a/src/profiler.cc
+++ b/src/profiler.cc
@@ -380,21 +380,21 @@ profile_data_accumulator::get_hierarchic
 
   return retval;
 }
 
 double
 profile_data_accumulator::query_time (void) const
 {
   octave_time now;
-  // FIXME -- this should be removed at some point...  See bug 34210.
-#if defined (__CYGWIN__) || defined (__MINGW32__)
-  volatile
-#endif
-    double dnow = now.double_value ();
+
+  // FIXME -- is this volatile declaration really needed?
+  // See bug #34210 for additional details.
+  volatile double dnow = now.double_value ();
+
   return dnow;
 }
 
 void
 profile_data_accumulator::add_current_time (void)
 {
   const double t = query_time ();
   assert (last_time >= 0.0 && last_time <= t);
